### PR TITLE
Release v1.1.0

### PR DESCRIPTION
**Pull request by amanojja-tibco**
## Description
This PR has mainly three changes
1. A new command called "generate-mock" added. It creates a Node App for user, using an Open-API 3.0/2.0 spec. NPM library "codegen-swagger-express" is used for that. Code of the library is customized by the TCAM team . Thus, it is stored under the assets folder of the repo.
 

2. Public APIs names and parameters changed as per requirements and suggestions provided by the BW team. These APIs are also used by the TCAM CLI , so our code is also updated according to the change.

3. Command flag names, description and examples updated.

## Motivation and Context
https://jira.tibco.com/browse/TCAM-211
https://jira.tibco.com/browse/TCAM-232

## How has this been tested?

Manually

## Types of changes

- [x] New command added
- [x] APIs updated
- [x] Command flag names updated
- [x] Command examples updated
- [x] Documentation related changes

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/commands/tcam/list-apis.ts [hunk 1]
```diff
@@ -3,75 +3,82 @@
...
     let params = new URLSearchParams();
-    if(flags.projectname){
-     const  projects = flags.projectname.split(',').map((name:any) => name.trim());
-    projects.forEach((proj:string) => {
-      params.append('projectNames',proj);
-    });
+    if (flags.projectnames) {
+      const projects = flags.projectnames.split(',').map((name: any) => name.trim());
+      projects.forEach((proj: string) => {
+        params.append('projectNames', proj);
```
Comment (fcastill) [inline src/commands/tcam/list-apis.ts:40]: this logic is repeated 2 more times for the other 2 flags. Consider extracting the logic to a reusable function.
I'm thinking something like this for a simple change:
```ts
parseListItems(flags.apitypes)
	.forEach((type) => params.append('apiTypes', type))
```

Or encapsulate all in a function and pass the params as parameter:

```ts
if (flag.apitypes) {
  addListParams(params, flags.apitypes, 'apiTypes');
}

//
function addListParams(params: URLSearchParams, flagListValue: string, paramName: string) {
  flagListValue
    .split(',')
    .forEach(item => {
       params.push(paramName, item.trim())
    })
    
}
```

**Review by fcastill — COMMENTED** [review]
Great organization. I added a couple comments for improvements.
I'd also like to discuss in more detail the strategy of including `code-swagger-express` the way it is currently included, the challenges you faced and the plan for maintaining that part of the codebase.

File: assets/mock-module/mock.handler.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  spec,
+  dynamic
+) => {
+  const oasType = spec.openapi ? OAS3 : OAS2;
+  // check if responses contains 200 response
+  if (responses[CODE_200]) {
+    const schema =
+      oasType === OAS3
+        ? responses[CODE_200].content ? responses[CODE_200].content["application/json"]?.schema : undefined
+        : responses[CODE_200].schema;
```
[the remark this answers is not in the document]
Comment (amanojja-tibco) [reply assets/mock-module/mock.handler.js]: Nested ternary removed. Now logic changed to only to single level.
Commit link: [](https://github.com/TIBCOSoftware/cic-cli-plugin-tcam/commit/df0e8675f2c45bf102d0a2099add907ce064677b)

File: assets/mock-module/commom.function.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const jsf = require("json-schema-faker");
```
[the remark this answers is not in the document]
Comment (amanojja-tibco) [reply assets/mock-module/commom.function.js]: File renamed to common.functions.js
Commit link: https://github.com/TIBCOSoftware/cic-cli-plugin-tcam/commit/df0e8675f2c45bf102d0a2099add907ce064677b

File: assets/mock-module/oas2Static.handler.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        return response;
+      }
+      const dummyExample = getDummyResponse(schema);
+      return `${JSON.stringify(dummyExample, null, 4)}`;
+    } else {
+      return "'200 okay'";
+    }
+  };
+}
+module.exports = new Oas2StaticHandler();
```
[the remark this answers is not in the document]
Comment (amanojja-tibco) [reply assets/mock-module/oas2Static.handler.js]: Class removed now. I have directly exported the functions as per suggestion,
Commmit link: https://github.com/TIBCOSoftware/cic-cli-plugin-tcam/commit/df0e8675f2c45bf102d0a2099add907ce064677b

File: assets/mock-module/oas3Static.handler.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        return response;
+      }
+      const dummyExample = getDummyResponse(schema);
+      return `${JSON.stringify(dummyExample, null, 4)}`;
+    } else {
+      return "'200 okay'";
+    }
+  };
+}
+module.exports = new Oas3StaticHandler();
```
[the remark this answers is not in the document]
Comment (amanojja-tibco) [reply assets/mock-module/oas3Static.handler.js]: Class removed now. I have directly exported the functions as per suggestion,
Commmit link: https://github.com/TIBCOSoftware/cic-cli-plugin-tcam/commit/df0e8675f2c45bf102d0a2099add907ce064677b

File: src/commands/tcam/create-project.ts [hunk 6]
```diff
@@ -25,14 +28,14 @@ export default class TcamCreateProject extends TCBaseCommand {
 
   async run() {
     const {flags} = this.parse(TcamCreateProject);
-    const projectname = flags.projectname.trim();
+    const projectname = flags.name.trim();
     let tcReq = this.getTCRequest();
     this.spinner.start("creating project...");
-    const res:any=await tcReq.doRequest(CLIAPIS.createproject,{method:'POST'},{projectName:projectname});
+    const res: any = await tcReq.doRequest(CLIAPIS.createproject,{method:'POST'},{projectName:projectname});
```
[the remark this answers is not in the document]
Comment (amanojja-tibco) [reply src/commands/tcam/create-project.ts]: any removed. Now, I have explicitly defined the type for TCRequest. This has solved the issue.
Commmit link: https://github.com/TIBCOSoftware/cic-cli-plugin-tcam/commit/df0e8675f2c45bf102d0a2099add907ce064677b

File: src/commands/tcam/generate-mock.ts [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+const DATE = Date.now();
+
+export default class TcamGenMock extends TCBaseCommand {
+  static description = 'Generate a NodeJS mock app';
+  static examples: string[] | undefined =
+    [`tibco tcam:generate-mock --from "C:/Users/myuser/Desktop/Upload/ImportApi.json" --deploy --static`,
+      `tibco tcam:generate-mock --project NodeProj --api Petstore --deploy`,
+      `tibco tcam:generate-mock -f "C:/Users/myuser/Desktop/Upload/ImportApi.json" -d -s`];
+  spinner: any;
+  tcReq: any;
```
[the remark this answers is not in the document]
Comment (amanojja-tibco) [reply src/commands/tcam/generate-mock.ts]: any removed. Now, I have explicitly defined the type for TCRequest. This has solved the issue.
Commmit link: https://github.com/TIBCOSoftware/cic-cli-plugin-tcam/commit/df0e8675f2c45bf102d0a2099add907ce064677b

File: src/commands/tcam/generate-mock.ts [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+const codegen = require('../../../assets/codegen-swagger-express');
+const DATE = Date.now();
+
+export default class TcamGenMock extends TCBaseCommand {
+  static description = 'Generate a NodeJS mock app';
+  static examples: string[] | undefined =
+    [`tibco tcam:generate-mock --from "C:/Users/myuser/Desktop/Upload/ImportApi.json" --deploy --static`,
+      `tibco tcam:generate-mock --project NodeProj --api Petstore --deploy`,
+      `tibco tcam:generate-mock -f "C:/Users/myuser/Desktop/Upload/ImportApi.json" -d -s`];
+  spinner: any;
```
[the remark this answers is not in the document]
Comment (amanojja-tibco) [reply src/commands/tcam/generate-mock.ts]: Thanks for telling the type for the spinner. I have now updated the type for spinner now.
Commmit link: https://github.com/TIBCOSoftware/cic-cli-plugin-tcam/commit/df0e8675f2c45bf102d0a2099add907ce064677b

File: src/commands/tcam/generate-mock.ts [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+          apiName: specName,
+          apiSpecType: 'openapi',
+          schemaType: 'OAS3.0',
+          content: JSON.stringify(spec),
+          apiVersion: '1.0',
+          apiContentType: 'json'
+        }
+      ];
+      const payload = { apiList: apiArr };
+      const res: any = await this.tcReq.doRequest(CLIAPIS.validateapis, { method: 'POST' }, payload);
```
[the remark this answers is not in the document]
Comment (amanojja-tibco) [reply src/commands/tcam/generate-mock.ts]: Resolved now, any removed.
Commmit link: https://github.com/TIBCOSoftware/cic-cli-plugin-tcam/commit/df0e8675f2c45bf102d0a2099add907ce064677b

File: src/commands/tcam/import-apis.ts [hunk 10]
```diff
@@ -40,56 +45,65 @@ export default class TcamImportApis extends TCBaseCommand {
   }
   async run() {
```
[the remark this answers is not in the document]
Comment (amanojja-tibco) [reply src/commands/tcam/import-apis.ts:48]: I have splitted the run function. To display the response of command , new function is added.
Commmit link: https://github.com/TIBCOSoftware/cic-cli-plugin-tcam/commit/df0e8675f2c45bf102d0a2099add907ce064677b

File: src/commands/tcam/list-apis.ts (hunk 1 above)
Comment (amanojja-tibco) [reply src/commands/tcam/list-apis.ts]: A common function called **_addListParams_** added. 
Coomit link: https://github.com/TIBCOSoftware/cic-cli-plugin-tcam/commit/df0e8675f2c45bf102d0a2099add907ce064677b

**Review by fcastill — APPROVED** [review]
Looks good to me!

**Pull request merged**